sequenceserver.gemspec

```diff
@@ -1,56 +1,58 @@
 require File.expand_path('lib/sequenceserver/version.rb',
                          File.dirname(__FILE__))
 Gem::Specification.new do |s|
-  # meta
-  s.name        = 'sequenceserver'
-  s.version     = SequenceServer::VERSION
-  s.authors     = ['Anurag Priyam', 'Ben J Woodcroft',
-                   'Vivek Rai', 'Yannick Wurm']
-  s.email       = ['anurag08priyam@gmail.com', 'b.woodcroft@uq.edu.au']
-  s.homepage    = 'http://sequenceserver.com'
-  s.license     = 'AGPL-3.0'
-
-  s.summary     = 'BLAST search made easy!'
-  s.description = <<~DESC
-    SequenceServer lets you rapidly set up a BLAST+ server with an intuitive
-    user interface for use locally or over the web.
-  DESC
-
-  # dependencies
-  s.required_ruby_version = '>= 2.3.0'
-
-  s.add_dependency('json_pure', '~> 1.8',  '>= 1.8.2')
-  s.add_dependency('ox',        '~> 2.1',  '>= 2.1.1')
-  s.add_dependency('sinatra',   '~> 2.0',  '>= 2.0.0')
-  s.add_dependency('slop',      '~> 3.6',  '>= 3.6.0')
-
-  s.add_development_dependency('capybara', '~> 3.15', '>= 3.15.1')
-  s.add_development_dependency('capybara-screenshot', '~> 1.0', '>= 1.0.23')
-  s.add_development_dependency('rack-test', '~> 1.0', '>= 1.0.0')
-  s.add_development_dependency('rspec', '~> 3.7', '>= 3.7.0')
-  s.add_development_dependency('rubocop', '~> 0.75', '>= 0.75.1')
-  s.add_development_dependency('selenium-webdriver', '~> 3.142', '>= 3.142.3')
-  s.add_development_dependency('simplecov', '~> 0.17.0', '>= 0.17.0')
-
-  # gem
-  s.files         = Dir['{bin,lib,public,views}/**/*'] +
-                     %w[config.ru COPYRIGHT.txt LICENSE]
-  s.executables   = ['sequenceserver']
-  s.require_paths = ['lib']
-
-  # post install information
-  s.post_install_message = <<~INFO
-
-    ------------------------------------------------------------------------
-      Thank you for installing SequenceServer :)
-
-      To launch SequenceServer execute 'sequenceserver' from command line.
-
-        $ sequenceserver
-
-
-      Visit http://sequenceserver.com for more.
-    ------------------------------------------------------------------------
-
-  INFO
+# meta
+s.name        = 'sequenceserver'
+s.version     = SequenceServer::VERSION
+s.authors     = ['Anurag Priyam', 'Ben J Woodcroft', 'Vivek Rai', 'Yannick Wurm']
+s.email       = ['anurag08priyam@gmail.com', 'b.woodcroft@uq.edu.au']
+s.homepage    = 'http://sequenceserver.com'
+s.license     = 'AGPL-3.0'
+
+s.summary     = 'BLAST search made easy!'
+s.description = <<~DESC
+  SequenceServer lets you rapidly set up a BLAST+ server with an intuitive
+  user interface for use locally or over the web.
+DESC
+
+# dependencies
+s.required_ruby_version = '>= 3.1.2'
```

Comment:
I think we need to distinguish between ensuring seqserv works on latest Ruby and the minimum Ruby version we support. As a rule of thumb, the minimum Ruby version we want to support is the default Ruby of the oldest Ubuntu LTS that has not entered ESM phase yet. That would be Ubuntu 18.04 and the default for Ubuntu 18.04 is Ruby 2.5. So we should ideally have `>= 2.5.0` here.

Reply:
Is it possible for us to support all the way from Ruby 2.5 to 3.1?

Reply:
I appreciate the reasoning that soon enough we will run into the next problem, but we should think from the user's point of view as well: Enforcing such a high minimum Ruby version constraint will make it harder for many existing users to upgrade and for new user to even try it out.

One way of approaching this is to test against multiple Ruby versions on Travis. Maybe just Ruby 2.5 and 3.1. That is, the minimum required and the latest. If installation and tests start failing for the minimum Ruby version and without a simple fix, we increase the minimum Ruby. It used to be the case earlier that we used to test against multiple Ruby versions. But there was a phase when our testing system went through a lot of changes. And it got too complex to test multiple Ruby versions while also experimenting with different approaches, so the number of Ruby versions to run the tests on was reduced. I think the test suite is in a much better position today (except 1-2 remaining random errors which I am sure we will get to as well).